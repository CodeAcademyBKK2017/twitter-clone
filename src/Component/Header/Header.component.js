import inputStyles from './Header.css';
import React, {Component} from 'react';
import CircleBG from '../Circle/CircleBG.component';
import CircleNoneBG from '../Circle/CircleNoneBG.component';
import InputBorder from '../InputBorder/InputBorder.component';
import Icon from 'react-ionicons';
import className from 'classname';

class Header extends Component {
  state = {
    menuVisibility: false
  }
  toggleMenu =() => {
    this.setState({menuVisibility: !this.state.menuVisibility});
  }
  render () {
    return (
      <div className={className(inputStyles.header, inputStyles.shadow)}>
        { 
          this.state.menuVisibility ?
            <div className={className(inputStyles.sidemenu)}>
              <div className={inputStyles.borderStyle}>
                <input className={inputStyles.inputStyle} placeholder="Search"/>
                <Icon className={inputStyles.colorStyle} icon='ios-search'/>
              </div>
              <div className={inputStyles.name}>Nanny</div>
            </div> : null
        }
        
        <div className={className(inputStyles.show)} onClick={this.toggleMenu}>
          <Icon icon="ios-menu"/></div> 
        <div className={className(inputStyles.flex4, inputStyles.contentLeft)}>
        
          <CircleBG iconName={'logo-twitter'} color='white'/>

          <div className={inputStyles.divide}/>

          <CircleNoneBG color='#5f5f5f' iconName={'ios-notifications'}/>
          <CircleNoneBG color='#5f5f5f' iconName={'ios-mail'}/>
          <CircleNoneBG color='#5f5f5f' iconName={'ios-grid-outline'}/>

        </div>

        <div className={inputStyles.flex1}><Icon fontSize='35px' color='#1e90ff' icon='logo-twitter'/></div>

        <div className={className(inputStyles.flex4, inputStyles.contentRight)}>
          <InputBorder />
          <CircleNoneBG color='red' iconName={'md-heart'}/>
        </div>
        
      </div>

    );
  }
}

export default Header;
