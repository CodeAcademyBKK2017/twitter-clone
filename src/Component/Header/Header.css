  .Box-Header {  
    display: flex;  
    height: 60px;
    position: fixed;
    top:0;
    width:100%;    
    justify-content: center;
    align-items: center;
  } 
  
  .Header4 {
    flex-grow: 4;
  } 
  
  .Header1 {
    flex-grow: 1;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .logo {
    color:#80A8CC;
    size : 2x;
  }
  
