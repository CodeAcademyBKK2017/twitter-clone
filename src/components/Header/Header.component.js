import HeaderStyle from './Header.style.css';
import LandingPageStyle from '../LandingPage/LandingPage.style.css';
import CircleStyle from '../Circle/Circle.style.css';
import React, {Component} from 'react';
import Circle from '../Circle/Circle.component';
import SearchBar from '../SearchBar/SearchBar.component';
import SideMenu from '../SideMenu/SideMenu.component';
import Icon from 'react-ionicons';

class Header extends Component {

	state = {
		isHideSideMenu: true
	};

	onMenuButtonClick = () => {
		const newState = {...this.state};
		newState.isHideSideMenu = !this.state.isHideSideMenu;
		this.setState(newState);
	}

	render () {
		return (
			<div className={`${LandingPageStyle.box} ${HeaderStyle.twitterHeader}`}>
				<div className={HeaderStyle.mainContainer}>
					<div className={HeaderStyle.leftContent}>
						<div className={HeaderStyle.brandContainer}>
							<Circle isBrand={true} iconName="logo-twitter"/>
						</div>
					
						<div className={HeaderStyle.lineSpliter}/>

						<div className={HeaderStyle.buttonContainer}>
							<Circle iconName="ios-notifications-outline"/>
							<Circle iconName="ios-mail-outline"/>
							<Circle iconName="ios-grid-outline"/>
							<Circle/>
						</div>
					</div>

					<div className={HeaderStyle.logoContent}>
						<Icon className={HeaderStyle.twitterLogo} icon="logo-twitter" fontSize="35px" color="#80A8CC"/>
					</div>

					<div className={HeaderStyle.rightContent}>
						<SearchBar/>

						<div className={HeaderStyle.brandContainer}>
							<Circle>
								<img className={`${CircleStyle.circle} ${CircleStyle.image}`} src="https://cdn1.iconfinder.com/data/icons/ninja-things-1/1772/ninja-simple-512.png" alt=""/>
							</Circle>
						</div>
					</div>
				</div>

				<div className={HeaderStyle.mobileContainer}>
					<div className={HeaderStyle.leftContent}>
						<div className={HeaderStyle.brandContainer}>
							<Circle iconName="ios-menu" onButtonClick={this.onMenuButtonClick}/>
						</div>
					</div>

					<div className={HeaderStyle.logoContent}>
						<Icon className={HeaderStyle.twitterLogo} icon="logo-twitter" fontSize="35px" color="#80A8CC"/>
					</div>

					<div className={HeaderStyle.rightContent} />

					<SideMenu isHidden={this.state.isHideSideMenu}/>
				</div>
			</div>
		);
	}
}

export default Header;
