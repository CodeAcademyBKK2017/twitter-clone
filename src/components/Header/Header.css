.background_white{
    background-color: white;
}

.box_shadow {
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.08), 0 0 0 1px rgba(0,0,0,0.04);
}

.header {
    position: fixed;
    left: 0;
    right: 0;
    height: 60px;
    top: 0;
    display: flex;
}

.left_head{
    flex: 4;
    display: flex;
    align-items: center;
    justify-content: flex-start;
}
.center_head{
    flex: 1;
    color: #80A8CC;
    display: flex;
    align-items: center;
    justify-content: center;
}
.right_head{
    flex: 4;
    display: flex;
    align-items: center;
    justify-content: flex-end;
}
.right_head_text {
    padding: 5px;
    border: 1px solid gray;
    display: inline-flex;
    border-radius: 10px;
}
.right_head_text > input {
    border: none;
    outline: none;
    background: none;
    width: 20vw;
}
.right_head_icon {
    padding-right: 10px;
}

.head_first_icon {
    padding: 0 10px 0 0;
    border-right: 1px solid black;
    height: 100%;
    align-items: center;
    display: flex;
}
.head_second_icon {
    height: 100%;
    align-items: center;
    display: flex;
}

.header_mobile {
    display: none;
}
.menuBar {
    left: -80vw;
}

@media only screen and (max-width: 768px) {
    .header {
        display: none;
    }
    .header_mobile {
        position: fixed;
        left: 0;
        right: 0;
        height: 60px;
        top: 0;
        display: flex;
    }

    .menuBar{
        /* display: inline-flex; */
        bottom: 0;
        top: 60px;
        position: fixed;
        /* width: 80vw; */
        background-color: #F2F2F2;
        padding: 5px 5px 5px 5px;
        transition: all 0.5s;
    }
    .searchTextBox{
        /* display: inline-flex; */
        width: 100%;
    }
    .right_head_text > input{
        /* display: inline-flex; */
        width: 100%;
    }
    .bottomText{
        position: absolute;
        bottom: 0;
        left: 0;
        right: 0;
        text-align: center;
        padding: 10px;
        border-top: 2px solid #ccc;
    }

    .activeMenu{
        left: 0vw;
        width: 80vw;
    }
}