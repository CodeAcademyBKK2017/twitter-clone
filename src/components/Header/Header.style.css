.header {
    position: fixed;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    height: 60px;
    background-color: white;
    color: black;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.08), 0 0 0 1px rgba(0,0,0,0.04);
    display: flex;
}
.left {
    flex: 4;
    display: flex;
    align-items: center;
}
.logo {
    flex: 1;
    height: 100%;
    color: #80A8CC;
    display: flex;
    align-items: center;
    justify-content: center;
}
.right {
    flex: 4;
    display: flex;
    align-items: center;
    justify-content: flex-end;
}
.circleContainer {
    height: 100%;
    display: inline-flex;
    align-items: center;
    justify-content: center;
}
.circleContainer__first {
    width: 55px;
    margin-right: 5px;
}
.circleContainer__user {
    width: 55px;
}
.circleContainer__borderRight {
    border-right: 1px solid #f8f4ee;
}
.circleContainer__borderBottom {
    border-bottom: 2px solid #81a9cb;
}
.circleContainer+.circleContainer {
    width: 45px;
}