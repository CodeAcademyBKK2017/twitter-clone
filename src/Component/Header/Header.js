import './Header.css';
import Circle from '../Circle/Circle'
import React, {Component} from 'react';
import {Icon} from 'react-fa';

class Header extends Component {
  render () {
    return (     
        <div className='Box-Header shadow box-bg'>
          <div className='Header4'><Circle name='twitter'/></div>
          <div className='Header1'><Icon size='2x' className='logo' name="twitter"/></div>
          <div className='Header4'>c</div>        
        </div>      
    );
  }
}

export default Header;
