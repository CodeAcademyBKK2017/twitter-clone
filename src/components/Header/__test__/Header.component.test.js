import Header from '../Header.component';
import React from 'react';
import renderer from 'react-test-renderer';
describe('Header group Test', () => {
  it('Header: snapshot test', () => {
    const snapshot = renderer.create(<Header />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});