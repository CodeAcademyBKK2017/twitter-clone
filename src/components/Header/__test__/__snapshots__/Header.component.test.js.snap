// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header group Test Header: snapshot test 1`] = `
<div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className="undefined undefined"
        >
           
          <svg
            className="sc-bdVaJa fUuvxv"
            fill="white"
            height="22px"
            onClick={undefined}
            rotate={0}
            style={Object {}}
            viewBox="0 0 1024 1024"
            width="22px"
          >
            <path
              d="M984 219c-34.8 15.4-72 25.8-111.2 30.6 40-24 70.8-62 85.2-107.2-37.4 22.2-78.8 38.4-123 47-35.4-37.8-85.8-61.4-141.4-61.4-107 0-193.6 86.8-193.6 193.8 0 15.2 1.6 30 5 44.2-161-8-303.8-85.2-399.2-202.6-16.6 28.6-26.2 62-26.2 97.4 0 67.2 34.4 126.6 86.4 161.4-32-0.8-62-9.6-88-24.2 0 0.8 0 1.6 0 2.4 0 94 66.8 172.2 155.4 190-16.2 4.4-33.4 6.8-51 6.8-12.4 0-24.6-1.2-36.4-3.6 24.6 77 96.2 133 181 134.6-66.2 52-149.8 83-240.6 83-15.6 0-31-1-46.2-2.8 85.4 55.6 187.2 87.6 296.4 87.6 356.6 0 551.4-295.4 551.4-551.6 0-8.4-0.2-16.8-0.6-25 37.8-27.4 70.6-61.4 96.6-100.4z"
            />
          </svg>
           
        </div>
      </div>
      <div
        className="undefined undefined"
      >
         
        <svg
          className="sc-bdVaJa fUuvxv"
          fill="#000000"
          height="22px"
          onClick={undefined}
          rotate={0}
          style={Object {}}
          viewBox="0 0 1024 1024"
          width="22px"
        >
          <path
            d="M576 800c0 34-28.8 64-64 64s-64.4-30-64.4-64h-32c0 52 43.2 95.8 96.2 95.8s96.2-43.8 96.2-95.8h-32z M747.8 458.4c0-198-98-265.6-189.8-281 0-1.2 0.2-2.2 0.2-3.4 0-25.4-20.6-46-46-46s-46 20.6-46 46c0 1.2 0 2.4 0.2 3.6-91.8 15.6-189.8 83.8-189.8 281.4-0.4 231-45.4 245-116.6 309h704c-70.8-64-116.2-78.2-116.2-309.6zM242.2 736c14.4-16 27.4-34.8 37.8-60 19.6-47.4 28.2-114.6 28.2-217.4 0-82.6 18-146.2 53.6-189 27.2-32.8 67.4-55.4 110.8-61 33.6-4.2 44.4-4.8 79.4 0 41.6 5.8 83.2 28 110.4 60.8 35.4 42.8 53.4 106.4 53.4 189 0 102.8 8.8 169.8 28.4 217.4 10.4 25.2 23.4 44.2 38 60.2h-540z"
          />
        </svg>
         
      </div>
      <div
        className="undefined undefined"
      >
         
        <svg
          className="sc-bdVaJa fUuvxv"
          fill="#000000"
          height="22px"
          onClick={undefined}
          rotate={0}
          style={Object {}}
          viewBox="0 0 1024 1024"
          width="22px"
        >
          <path
            d="M128 256v512h768v-512h-768zM512 535.8l-325.6-247.8h651.2l-325.6 247.8zM160 736v-427.8l230.2 175.2-136.2 154.6 4 4 157.8-139.2 96.2 73.2 96.2-73.2 157.8 139.2 4-4-136.2-154.8 230.2-175v427.8h-704z"
          />
        </svg>
         
      </div>
      <div
        className="undefined undefined"
      >
         
        <svg
          className="sc-bdVaJa fUuvxv"
          fill="#000000"
          height="22px"
          onClick={undefined}
          rotate={0}
          style={Object {}}
          viewBox="0 0 1024 1024"
          width="22px"
        >
          <path
            d="M896 384v-32h-224v-224h-32v224h-256v-224h-32v224h-224v32h224v256h-224v32h224v224h32v-224h256v224h32v-224h224v-32h-224v-256h224zM640 640h-256v-256h256v256z"
          />
        </svg>
         
      </div>
    </div>
    <div
      className={undefined}
    >
       
      <svg
        className="sc-bdVaJa fUuvxv"
        fill="#00aced"
        height="22px"
        onClick={undefined}
        rotate={0}
        style={Object {}}
        viewBox="0 0 1024 1024"
        width="22px"
      >
        <path
          d="M984 219c-34.8 15.4-72 25.8-111.2 30.6 40-24 70.8-62 85.2-107.2-37.4 22.2-78.8 38.4-123 47-35.4-37.8-85.8-61.4-141.4-61.4-107 0-193.6 86.8-193.6 193.8 0 15.2 1.6 30 5 44.2-161-8-303.8-85.2-399.2-202.6-16.6 28.6-26.2 62-26.2 97.4 0 67.2 34.4 126.6 86.4 161.4-32-0.8-62-9.6-88-24.2 0 0.8 0 1.6 0 2.4 0 94 66.8 172.2 155.4 190-16.2 4.4-33.4 6.8-51 6.8-12.4 0-24.6-1.2-36.4-3.6 24.6 77 96.2 133 181 134.6-66.2 52-149.8 83-240.6 83-15.6 0-31-1-46.2-2.8 85.4 55.6 187.2 87.6 296.4 87.6 356.6 0 551.4-295.4 551.4-551.6 0-8.4-0.2-16.8-0.6-25 37.8-27.4 70.6-61.4 96.6-100.4z"
        />
      </svg>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <input
            className={undefined}
          />
          <div
            className={undefined}
          >
            <svg
              className="sc-bdVaJa fUuvxv"
              fill="#000000"
              height="22px"
              onClick={undefined}
              rotate={0}
              style={Object {}}
              viewBox="0 0 1024 1024"
              width="22px"
            >
              <path
                d="M689 596c30-47.2 47.6-103.2 47.6-163.4 0-168.2-136.2-304.6-304.2-304.6-168.2 0-304.4 136.4-304.4 304.6s136.2 304.6 304.2 304.6c61 0 117.8-18 165.4-48.8l13.8-9.6 217.2 217.2 67.4-68.6-217-217.2 10-14.2zM602.8 262.4c45.4 45.4 70.4 105.8 70.4 170s-25 124.6-70.4 170c-45.4 45.4-105.8 70.4-170 70.4s-124.6-25-170-70.4c-45.4-45.4-70.4-105.8-70.4-170s25-124.6 70.4-170c45.4-45.4 105.8-70.4 170-70.4s124.6 25 170 70.4z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="undefined undefined"
      >
         
        <img
          src="http://icons.iconarchive.com/icons/designbolts/despicable-me-2/32/Minion-Amazed-icon.png"
        />
         
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <svg
        className="sc-bdVaJa fUuvxv"
        fill="#00aced"
        height="22px"
        onClick={undefined}
        rotate={0}
        style={Object {}}
        viewBox="0 0 1024 1024"
        width="22px"
      >
        <path
          d="M128 288h768v64h-768v-64z M128 480h768v64h-768v-64z M128 672h768v64h-768v-64z"
        />
      </svg>
    </div>
    <div
      className={undefined}
    >
      <svg
        className="sc-bdVaJa fUuvxv"
        fill="#00aced"
        height="22px"
        onClick={undefined}
        rotate={0}
        style={Object {}}
        viewBox="0 0 1024 1024"
        width="22px"
      >
        <path
          d="M984 219c-34.8 15.4-72 25.8-111.2 30.6 40-24 70.8-62 85.2-107.2-37.4 22.2-78.8 38.4-123 47-35.4-37.8-85.8-61.4-141.4-61.4-107 0-193.6 86.8-193.6 193.8 0 15.2 1.6 30 5 44.2-161-8-303.8-85.2-399.2-202.6-16.6 28.6-26.2 62-26.2 97.4 0 67.2 34.4 126.6 86.4 161.4-32-0.8-62-9.6-88-24.2 0 0.8 0 1.6 0 2.4 0 94 66.8 172.2 155.4 190-16.2 4.4-33.4 6.8-51 6.8-12.4 0-24.6-1.2-36.4-3.6 24.6 77 96.2 133 181 134.6-66.2 52-149.8 83-240.6 83-15.6 0-31-1-46.2-2.8 85.4 55.6 187.2 87.6 296.4 87.6 356.6 0 551.4-295.4 551.4-551.6 0-8.4-0.2-16.8-0.6-25 37.8-27.4 70.6-61.4 96.6-100.4z"
        />
      </svg>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <input
            className={undefined}
          />
          <div
            className={undefined}
          >
            <svg
              className="sc-bdVaJa fUuvxv"
              fill="#000000"
              height="22px"
              onClick={undefined}
              rotate={0}
              style={Object {}}
              viewBox="0 0 1024 1024"
              width="22px"
            >
              <path
                d="M689 596c30-47.2 47.6-103.2 47.6-163.4 0-168.2-136.2-304.6-304.2-304.6-168.2 0-304.4 136.4-304.4 304.6s136.2 304.6 304.2 304.6c61 0 117.8-18 165.4-48.8l13.8-9.6 217.2 217.2 67.4-68.6-217-217.2 10-14.2zM602.8 262.4c45.4 45.4 70.4 105.8 70.4 170s-25 124.6-70.4 170c-45.4 45.4-105.8 70.4-170 70.4s-124.6-25-170-70.4c-45.4-45.4-70.4-105.8-70.4-170s25-124.6 70.4-170c45.4-45.4 105.8-70.4 170-70.4s124.6 25 170 70.4z"
              />
            </svg>
          </div>
        </div>
         
      </div>
      <div
        className={undefined}
      >
         Made By : Ben
      </div>
    </div>
  </div>
</div>
`;
